docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,282 @@
+
+### How the configuration is applied
+
+Grafana Mimir components are run with a configuration calculcated by the following process:
+
+1. The configuration YAML in `mimir.config` is evaulated for Helm templates. This step is intended to ensure that the configuration applies to the Kubernetes cluster where it will be installed. For example setting up cluster specific addresses.
+1. The values from `mimir.structuredConfig` are merged on top and the result is again evaulated for Helm templates. This steps is intended to apply user specific customizations. For example S3 storage details.
+1. The resulting YAML configuration is then sorted alphabetically and stored in a `ConfigMap` (or `Secret` depending on the value of `configStorageType`) and provided to all Grafana Mimir components.
+1. When Grafana Mimir components are run in pods, the configuration file as well as any extra CLI flags are provided to the component.
+1. Each component evaulates the configuration, substituting environment variables as required. Note that extra CLI flags take precedence over the configuration file.
+
+> **Note:**: CLI flags are component specific, thus they will not show up in the generated `ConfigMap` (or `Secret`), making it less obvious what configuration is running. Use only when absolutely necessary.
+
+### Inspect changes to the configuration before upgrade
+
+Follow these steps to inspect what change will be applied to the configuration.
+
+Preparation:
+
+1. Install the [helm diff](https://github.com/databus23/helm-diff) plugin.
+1. Make sure to use `configStorageType` is set to `ConfigMap`
+
+Inspecting changes with the `helm diff` sub command:
+
+```bash
+helm -n mimir-test diff upgrade grafana/mimir-distributed -f custom.yaml
+```
+
+This command shows the differences between the running deployment and the deployment that would result from executing the `helm upgrade` command. Search for `name: mimir-config` in the output to see the difference in configuration settings.
+
+> **Note:** CLI flags and their difference are found in the `Deployment` and `StatefulSet` objects.
+
+## Manage the configuration directly
+
+Prepare the configuration as text. It can not include Helm template functions, value evaulations. The configuration may include references to environment variables as explained in [Injecting credentials](#injecting-credentials).
+
+Decide whether you want to use a `ConfigMap` or `Secret` to store the configuration. Handling `ConfigMap` is a little bit simpler, but be ware of sensitive information.
+
+### Use external ConfigMap
+
+Prepare a `ConfigMap` object where the configuration is placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: |
+    <configuration>
```

Comment:
nit: a small example would be cool

```suggestion
    blocks_storage:
      backend: s3
      s3:
        endpoint: https://example.com
```

Reply:
this would also make redundant the following sentence

Reply:
ugh, that's not that simple , here you need to configure everything so I think giving just a few lines would be misleading

Reply:
i thought about this again. I think having the placeholder there is ok